homeassistant/components/nina/config_flow.py

```diff
@@ -0,0 +1,163 @@
+
+            errors["base"] = "no_selection"
+
+        self.split_regions()
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(CONF_REGIONS + "1"): cv.multi_select(self.regions_a),
+                    vol.Optional(CONF_REGIONS + "2"): cv.multi_select(self.regions_b),
+                    vol.Optional(CONF_REGIONS + "3"): cv.multi_select(self.regions_c),
+                    vol.Optional(CONF_REGIONS + "4"): cv.multi_select(self.regions_d),
+                    vol.Optional(CONF_REGIONS + "5"): cv.multi_select(self.regions_e),
+                    vol.Optional(CONF_REGIONS + "6"): cv.multi_select(self.regions_f),
+                    vol.Required(CONF_MESSAGE_SLOTS, default=5): vol.All(
+                        int, vol.Range(min=1, max=20)
+                    ),
+                    vol.Required(CONF_FILTER_CORONA, default=True): cv.boolean,
+                }
+            ),
+            errors=errors,
+        )
+
+    @staticmethod
+    def swap_key_value(dict_to_sort: dict[str, str]) -> dict[str, str]:
+        """Swap keys and values in dict."""
+        all_region_codes_swaped: dict[str, str] = {}
+
+        for key, value in dict_to_sort.items():
+            if value not in all_region_codes_swaped:
+                all_region_codes_swaped[value] = key
+            else:
+                for i in range(len(dict_to_sort)):
+                    tmp_value: str = value + "_" + str(i)
+                    if tmp_value not in all_region_codes_swaped:
+                        all_region_codes_swaped[tmp_value] = key
+                        break
+
+        return dict(sorted(all_region_codes_swaped.items(), key=lambda ele: ele[1]))
+
+    def split_regions(self) -> None:
+        """Split regions alphabetical."""
+        for i, name in self._all_region_codes_sorted.items():
+            if name[0] == "A" or name[0] == "B" or name[0] == "C" or name[0] == "D":
+                self.regions_a[i] = name
+            if name[0] == "E" or name[0] == "F" or name[0] == "G" or name[0] == "H":
+                self.regions_b[i] = name
+            if name[0] == "I" or name[0] == "J" or name[0] == "K" or name[0] == "L":
+                self.regions_c[i] = name
+            if (
+                name[0] == "M"
+                or name[0] == "N"
+                or name[0] == "O"
+                or name[0] == "P"
+                or name[0] == "Q"
+            ):
+                self.regions_d[i] = name
+            if name[0] == "R" or name[0] == "S" or name[0] == "T" or name[0] == "U":
+                self.regions_e[i] = name
+            if (
+                name[0] == "V"
+                or name[0] == "W"
+                or name[0] == "X"
+                or name[0] == "Y"
+                or name[0] == "Z"
+            ):
+                self.regions_f[i] = name
```

Comment:
I get the idea behind this, can we maybe move all `regions_*` into a maybe `dict[str, dict[str, Any]]` as proposed above. Or literally anything that makes this more succint.

You might then maybe want to define lists for your groups in `const.py` and simply use:

```python
for i, name in self._all_region_codes_sorted.items():
            if name[0] in CONST_LIST_A_TO_D:
```

Reply:
Since this is a german API: Are we sure region names do not start with Ä Ö or any other umlaut?

Haven't checked that while testing locally.